tests/test_cmab.py

```
@@ -569,6 +590,37 @@ def test_bad_initialization(
+        num_models = 1
+        if issubclass(type(action), ModelMO):
+            curr_action = action.models[0]
+            num_models = len(action.models)
+        else:
+            curr_action = action
+        if issubclass(type(curr_action), BaseBayesianNeuralNetwork):
+            model_n_features = n_features
+        elif issubclass(type(curr_action), BaseQuantitativeBayesianNeuralNetwork):
+            model_n_features = n_features + 1
+        else:
+            raise ValueError(f"Invalid action type: {type(action)}")
```

Ruff (0.15.2)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)